moped-database/migrations/1723496266686_add_bike_share_station_comp/up.sql

```diff
@@ -0,0 +1,37 @@
+-- add bike share station component type (point) to moped_components table if it doesn't already exist
+INSERT INTO moped_components (component_name, line_representation, feature_layer_id)
+SELECT
+    'Bike Share Station',
+    FALSE,
+    5
+WHERE NOT EXISTS (
```

Comment:
this was a TIL for me to check for an existing value before inserting. I have only used `ON CONFLICT` to do something like this before.

Reply:
+1 — good to know of this approch!

Reply:
why do we need to check if the component already exists in the table? just so that if we run the down migration and then up migration again we wouldnt double insert?

Reply:
yep - that is exactly why! thinking about how to avoid inserting twice became a fun detour.